app/models/PaypalApiError.scala

```diff
@@ -0,0 +1,39 @@
+package models
+
+import com.paypal.api.payments.Error
+import enumeratum.{Enum, EnumEntry, PlayJsonEnum}
+import play.api.libs.json.{JsValue, Json, Writes}
+
+
+sealed trait PaypalErrorType extends EnumEntry
+
+object PaypalErrorType extends Enum[PaypalErrorType] with PlayJsonEnum[PaypalErrorType] {
+  val values = findValues
+
+  case object NotFound extends PaypalErrorType
+  case object PaymentAlreadyDone extends PaypalErrorType
+  case object Other extends PaypalErrorType
+
+  def fromPaypalError(error: Error): PaypalErrorType = error.getName match {
+    case "INVALID_RESOURCE_ID" => NotFound
+    case "PAYMENT_ALREADY_DONE" => PaymentAlreadyDone
+    case _ => Other
+  }
+}
+
+case class PaypalApiError(
+  errorType: PaypalErrorType,
+  message: String
+)
+
+object PaypalApiError {
+  def apply(message: String): PaypalApiError = PaypalApiError(PaypalErrorType.Other, message)
```

Comment:
Yeah I know, but I was hesitating between easy use of the PaypalApiError class, or writing a two field Writer. I'll use the fromString function instead